### PR TITLE
netpol: Use kube-router as a library

### PR DESCRIPTION
**Pull request by manuelbuil**
Signed-off-by: Michal Rostecki <mrostecki@opensuse.org>
Signed-off-by: Manuel Buil <mbuil@suse.com>

#### Proposed Changes ####

Before this change, we were copying a part of kube-router code to pkg/agent/netpol directory with modifications, from which the biggest one was consumption of k3s node config instead of kube-router config.

However, that approach made it hard to follow new upstream versions. It's possible to use kube-router as a library, so it seems like a better way to do that.

Instead of modifying kube-router network policy controller to comsume k3s configuration, this change just converts k3s node config into kube-router config. All the functionality of kube-router except netpol is still disabled.

#### Types of Changes ####

Code change to facilitate maintenance

#### Verification ####

Test network policies and they should work as they were working before

#### Linked Issues ####

https://github.com/k3s-io/k3s/issues/5078

#### User-Facing Change ####

```release-note

```

### CODE REVIEW

File: pkg/agent/netpol/netpol.go [hunk 1]
```diff
@@ -44,15 +51,37 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
...
+	krConfig.ClusterIPCIDR = nodeConfig.AgentConfig.ServiceCIDR.String()
+	krConfig.NodePortRange = strings.ReplaceAll(nodeConfig.AgentConfig.ServiceNodePortRange.String(), "-", ":")
+	krConfig.HostnameOverride = nodeConfig.AgentConfig.NodeName
+	krConfig.MetricsEnabled = false
+	krConfig.RunFirewall = true
+	krConfig.RunRouter = false
+	krConfig.RunServiceProxy = false
+
 	stopCh := ctx.Done()
+	healthCh := make(chan *healthcheck.ControllerHeartbeat, healthControllerChannelLength)
```
Comment (briandowns) [inline pkg/agent/netpol/netpol.go:64]: Is there a requirement to have this channel buffered?

File: pkg/agent/netpol/netpol.go [hunk 2]
```diff
@@ -44,15 +51,37 @@ func Run(ctx context.Context, nodeConfig *config.Node) error {
...
 	informerFactory.WaitForCacheSync(stopCh)
 
-	npc, err := NewNetworkPolicyController(client, nodeConfig, podInformer, npInformer, nsInformer, &sync.Mutex{})
+	// Start kube-router healthcheck server. Netpol requires it
+	hc, err := healthcheck.NewHealthController(krConfig)
+	if err != nil {
+		return err
+	}
+	wg.Add(1)
+	go hc.RunCheck(healthCh, stopCh, &wg)
```
Comment (briandowns) [inline pkg/agent/netpol/netpol.go:82]: The controller handles the waitgroup decrement?

**Review by brandond — APPROVED** [review]
LGTM. It appears that the controllers themselves decrement the waitgroup when they exit, so having the setup code add to the waitgroup [as upstream does](https://github.com/cloudnativelabs/kube-router/blob/ee9f6d890d10609284098229fa1e283ab5d83b93/pkg/cmd/kube-router.go#L202-L205) (even though we don't ever actually wait on it) looks like the correct thing to do to avoid panics on shutdown.

File: pkg/agent/netpol/netpol.go (hunk 2 above)
Comment (manuelbuil) [reply pkg/agent/netpol/netpol.go:82]: Yes! https://github.com/cloudnativelabs/kube-router/blob/master/pkg/healthcheck/health_controller.go#L186

File: pkg/agent/netpol/netpol.go (hunk 1 above)
Comment (manuelbuil) [reply pkg/agent/netpol/netpol.go:64]: Nope, let me remove it, I am not sure why we buffered it

**Review by briandowns — APPROVED** [review]
